swap_meet/vendor.py

```diff
@@ -1,2 +1,96 @@
+from queue import PriorityQueue
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory  = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            item not in self.inventory
+            return False
+    
+    def get_by_category(self,category):
+        category_list= []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+    
+    def swap_items(self, other_vendor, my_item,their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        if my_item in self.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.add(my_item)
+            self.inventory.append(their_item)
+            other_vendor.remove(their_item)
+            return True
+        
+#Wave 4
+    def swap_first_item(self, other_vendor):
+        if len(self.inventory) < 1 or len(other_vendor.inventory) < 1:
+            return False
+        else:
+            return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+
+    def get_best_by_category(self, category):
+        best_list = self.get_by_category(category)
+        best_item = ""
+        best_rating = 0
+        if len(best_list) < 1:
+            return None
+        for item in best_list:
+            if item.condition > best_rating:
+                best_item = item
+                best_rating = item.condition
+        return best_item
+        
+    def swap_best_by_category(self, other, my_priority, their_priority):
+            other_item = other.get_best_by_category(my_priority)
+            my_item = self.get_best_by_category(their_priority)
+            if my_item == None or other_item == None:
+                return False
+            else:
```

Comment:
Great job with this method, way to reuse all the previous methods! The `else` block can be moved into the main method body.